new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java

```diff
@@ -0,0 +1,1284 @@
+ * Also verify the sample application can be accessed via NGINX ingress controller.
+ */
+@DisplayName("Verify scaling the clusters in the domain with different domain types and "
+    + "the sample application can be accessed via NGINX ingress controller")
+@IntegrationTest
+class ItParameterizedDomain {
+
+  // domain constants
+  private static final int NUMBER_OF_CLUSTERS_MIIDOMAIN = 2;
+  private static final String CLUSTER_NAME_PREFIX = "cluster-";
+  private static final String clusterName = "cluster-1";
+  private static final int MANAGED_SERVER_PORT = 8001;
+  private static final int ADMIN_SERVER_PORT = 7001;
+  private static final int replicaCount = 2;
+  private static final String SAMPLE_APP_CONTEXT_ROOT = "sample-war";
+  private static final String WLDF_OPENSESSION_APP = "opensessionapp";
+  private static final String WLDF_OPENSESSION_APP_CONTEXT_ROOT = "opensession";
+  private static final String wlSecretName = "weblogic-credentials";
+
+  private static String wlsBaseImage = WLS_BASE_IMAGE_NAME + ":" + WLS_BASE_IMAGE_TAG;
+  private static String opNamespace = null;
+  private static String opServiceAccount = null;
+  private static HelmParams nginxHelmParams = null;
+  private static int nodeportshttp = 0;
+  private static int externalRestHttpsPort = 0;
+  private static boolean isUseSecret = true;
+  private static List<Domain> domains = new ArrayList<>();
+  private static LoggingFacade logger = null;
+
+  private String curlCmd = null;
+
+  /**
+   * Install operator and NGINX.
+   * Create three different type of domains: model in image, domain in PV and domain in image.
+   * Create ingress for each domain.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(5) List<String> namespaces) {
+    logger = getLogger();
+
+    // get a unique operator namespace
+    logger.info("Get a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    // get a unique NGINX namespace
+    logger.info("Get a unique namespace for NGINX");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    String nginxNamespace = namespaces.get(1);
+
+    // get unique namespaces for three different type of domains
+    logger.info("Getting unique namespaces for three different type of domains");
+    assertNotNull(namespaces.get(2));
+    String miiDomainNamespace = namespaces.get(2);
+    assertNotNull(namespaces.get(3));
+    String domainInPVNamespace = namespaces.get(3);
+    assertNotNull(namespaces.get(4));
+    String domainInImageNamespace = namespaces.get(4);
+
+    // set the service account name for the operator
+    opServiceAccount = opNamespace + "-sa";
+
+    // get a free port for external REST HTTPS port
+    externalRestHttpsPort = getNextFreePort(31001, 31201);
```

Comment:
You can use 0 for external rest port as well. I did the same in upgrade tests - https://github.com/oracle/weblogic-kubernetes-operator/blob/operator-upgrade-tests/new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java

Reply:
you can make this change in your next PR if you want.

Reply:
fixed